components/branding-preference-mgt/org.wso2.carbon.identity.branding.preference.mgt/src/main/java/org/wso2/carbon/identity/branding/preference/mgt/BrandingPreferenceManagerImpl.java

```diff
@@ -0,0 +1,324 @@
+                throw handleClientException(ERROR_CODE_BRANDING_PREFERENCE_NOT_EXISTS, tenantDomain);
+            }
+
+            InputStream inputStream = getConfigurationManager().getFileById
+                    (BRANDING_RESOURCE_TYPE, resourceName, resourceFiles.get(0).getId());
+            if (inputStream == null) {
+                throw handleClientException(ERROR_CODE_BRANDING_PREFERENCE_NOT_EXISTS, tenantDomain);
+            }
+            if (log.isDebugEnabled()) {
+                log.debug("Branding preference for tenant: " + tenantDomain + " is retrieved successfully.");
+            }
+            return buildBrandingPreferenceFromResource(inputStream, type, name, locale);
+        } catch (ConfigurationManagementException e) {
+            if (RESOURCE_NOT_EXISTS_ERROR_CODE.equals(e.getErrorCode())) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Can not find a branding preference configurations for tenant: " + tenantDomain, e);
+                }
+                throw handleClientException(ERROR_CODE_BRANDING_PREFERENCE_NOT_EXISTS, tenantDomain);
+            }
+            throw handleServerException(ERROR_CODE_ERROR_GETTING_BRANDING_PREFERENCE, tenantDomain, e);
+        } catch (IOException e) {
+            throw handleServerException(ERROR_CODE_ERROR_BUILDING_BRANDING_PREFERENCE, tenantDomain);
+        }
+    }
+
+    @Override
+    public BrandingPreference replaceBrandingPreference(BrandingPreference brandingPreference)
+            throws BrandingPreferenceMgtException {
+
+        String resourceName = getResourceName
+                (brandingPreference.getType(), brandingPreference.getName(), brandingPreference.getLocale());
+        String tenantDomain = getTenantDomain();
+        // Check whether the branding resource exists in the particular tenant.
+        if (!isResourceExists(BRANDING_RESOURCE_TYPE, resourceName)) {
+            throw handleClientException(ERROR_CODE_BRANDING_PREFERENCE_NOT_EXISTS, tenantDomain);
+        }
+
+        String preferencesJSON = generatePreferencesJSONFromPreference(brandingPreference.getPreference());
+        if (!BrandingPreferenceMgtUtils.isValidJSONString(preferencesJSON)) {
+            throw handleClientException(ERROR_CODE_INVALID_BRANDING_PREFERENCE, tenantDomain);
+        }
+
+        try {
+            InputStream inputStream = BrandingPreferenceMgtUtils.generatePreferenceInputStream(preferencesJSON);
+            Resource brandingPreferenceResource = buildResourceFromBrandingPreference(resourceName, inputStream);
+            getConfigurationManager().replaceResource(BRANDING_RESOURCE_TYPE, brandingPreferenceResource);
+        } catch (ConfigurationManagementException e) {
+            throw handleServerException(ERROR_CODE_ERROR_UPDATING_BRANDING_PREFERENCE, tenantDomain, e);
+        } catch (UnsupportedEncodingException e) {
+            throw handleServerException(ERROR_CODE_UNSUPPORTED_ENCODING_EXCEPTION, e);
```

Comment:
Shouldn't this be a client exception?

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/pull/3870/commits/d8b98f955a8d30d3fb07768ec4b1e5503aa46a48